src/lib/statistics/cardinality_estimator.cpp

```diff
@@ -417,13 +417,138 @@ std::shared_ptr<TableStatistics> CardinalityEstimator::estimate_predicate_node(
   const auto operator_scan_predicates = OperatorScanPredicate::from_expression(*predicate, predicate_node);
 
   // TODO(anybody) Complex predicates are not processed right now and statistics objects are forwarded.
-  //               That implies estimating a selectivity of 1 for such predicates
+  //               That implies estimating a selectivity of 1 for such predicates.
   if (!operator_scan_predicates) {
-    return input_table_statistics;
+    // We might not have resolved the predicate because it contains subqueries. Unfortunately, we do not know the
+    // values of predicates on uncorrelated subquery results before query execution. However, if the predicate has an
+    // equals or between condition, it acts as a filter comparable to a semi-join with the join key of the subquery
+    // result (see examples below). We obtain such predicates with subquery results from the JoinToPredicateRewriteRule.
+    // This rule also checks that all preconditions are met to ensure correct query results. Thus, we do not check them
+    // here. For more information about this query rewrite, see `join_to_predicate_rewrite_rule.hpp`.
```

Comment:
I haven't found text that explains that we only run into this branch here for the rewrite and not when somebody manually created "equal" subqueries.
I think that would help understanding this branch a lot.

Reply:
For the `column = <subquery>` case, we can actually have it triggered by the user with the example below:
```sql
SELECT n_name FROM nation WHERE n_regionkey = (SELECT r_regionkey FROM region WHERE r_name = 'ASIA');
```

However, the `SubqueryToJoinRule` eventually rewrites the subquery scan to a join.

What do you think of pulling the part starting from `We obtain such predicates ...` up and move it after the first sentence?

Reply:
Actually, I would even move it up to line 421.